meteor/scale.py

```diff
@@ -1,15 +1,222 @@
+from typing import Tuple
+
+import numpy as np
+import pandas as pd
 import reciprocalspaceship as rs
+import scipy.optimize as opt
+
+DTYPES_TO_SCALE = (
+    rs.AnomalousDifferenceDtype,
+    rs.FriedelIntensityDtype,
+    rs.FriedelStructureFactorAmplitudeDtype,
+    rs.IntensityDtype,
+    rs.NormalizedStructureFactorAmplitudeDtype,
+    rs.StandardDeviationDtype,
+    rs.StandardDeviationFriedelIDtype,
+    rs.StandardDeviationFriedelSFDtype,
+    rs.StructureFactorAmplitudeDtype,
+)
+""" automatically scale these types when they appear in an rs.DataSet """
+
+
+ScaleParameters = Tuple[float, float, float, float, float, float, float]
+""" 7x float tuple to hold anisotropic scaling parameters """
+
+
+def _compute_anisotropic_scale_factors(
+    miller_indices: pd.Index, anisotropic_scale_parameters: ScaleParameters
+) -> np.ndarray:
+    miller_indices_as_array = np.array(list(miller_indices))
+    squared_miller_indices = np.square(miller_indices_as_array)
+
+    h_squared = squared_miller_indices[:, 0]
```

Comment:
in #13 this block of code appears twice:

https://github.com/alisiafadini/meteor/blob/afc42ae62f8a65161eeaf60ff8bf293d43fc75c6/meteor/scale.py#L70

and

https://github.com/alisiafadini/meteor/blob/afc42ae62f8a65161eeaf60ff8bf293d43fc75c6/meteor/scale.py#L117

here it is enclosed in a function so that it appears once -- fair point that a more concise implementation might be possible, but I do like the explicitness of this version.